include/translatable-objects.php

```diff
@@ -37,6 +37,14 @@ class PLL_Translatable_Objects implements IteratorAggregate {
 	 *
 	 * @param PLL_Translatable_Object $object The translatable object to register.
 	 * @return PLL_Translatable_Object
+	 *
+	 * @phpstan-return (
+	 *     $object is PLL_Translated_Post ? PLL_Translated_Post : (
+	 *         $object is PLL_Translated_Term ? PLL_Translated_Term : (
+	 *             PLL_Translatable_Object
+	 *         )
+	 *     )
+	 * )
```

Comment:
I think using [generics](https://phpstan.org/blog/generics-in-php-using-phpdocs#type-variables) is a better option here:
- Not aware of implementations
- Work with other types than `PLL_Translated_Post` and `PLL_Translated_Term`
- No need to update the doc in case we add a new `PLL_Translatable_Object` implementation?

Reply:
Any idea @Screenfeed ?

Reply:
Well, as discussed there's no way to use generics...